### PR TITLE
feat(eslint): add linting rules for test describe depth

### LINKED ISSUE #122: [FEAT] Linting rules for describe() depth in tests
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[x] Feature request
[ ] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

## Current behavior
Currently, tests don't follow well defined patterns or style guides.

## Expected behavior
That tests follow well designed patterns and style guides.

## What is the motivation / use case for changing the behavior?
As a start, in addition to the Angular styleguide rules, it would be nice if we could enforce a "describe" depth of `3`, which would allow us to do meta-analysis of the complexity of a piece of code.

### PR DESCRIPTION
**Pull request by gracetxgao**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: #122

## What is the new behavior?

Maximum describe() depth of 3 set for tests.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

**Comment by damienwebdev** [comment]
This is awesome! @griest024 do you think we should set depth to 4 as warning, and 5 as error? I only see a few with 5, but many more with 3/4.

File: .eslintrc.js [hunk 1]
```diff
@@ -36,6 +36,12 @@ module.exports = {
             }
           }
         ],
+        "jest/max-nested-describe": [
+          "error",
+          {
+            "max": 3
+          }
+        ]
```
Comment (griest024) [inline .eslintrc.js]: This should go in the base config,: `tools/eslint/config/index.js`

**Review by griest024 — CHANGES_REQUESTED** [review]

**Comment by griest024** [comment]
@damienwebdev 5 as an error sounds good. Can we mix error and warning values for the same rule? Afaik eslint does not allow for this.

**Comment by damienwebdev** [comment]
https://github.com/graycoreio/daffodil/blob/d41bc5a949b1e827e35129ea4c0d449a2a8a7b47/tools/eslint/config/package.json#L14 needs an update for the dependency as well

**Review by damienwebdev — DISMISSED** [review]
lgtm. Needs @griest024 approval for merge.

File: tools/eslint/config/index.js [hunk 2]
```diff
@@ -15,6 +16,12 @@ module.exports = {
   ],
   rules: {
     // ng-cli-compat
+    "jest/max-nested-describe": [
+      "warn",
+      {
+        "max": 4
+      },
+    ],
```
Comment (griest024) [inline tools/eslint/config/index.js]: Move this under the `custom` section (denoted by the comment). Everything under `ng-cli-compat` and `ng-cli-compat--formatting-add-on` were added to replace deprecated packages and are going to eventually be mass deleted.

**Review by griest024 — CHANGES_REQUESTED** [review]

**Review by griest024 — DISMISSED** [review]
lgtm

**Comment by griest024** [comment]
needs a rebase

**Comment by damienwebdev** [comment]
@gracetxgao Needs another rebase.

**Review by damienwebdev — APPROVED** [review]
lgtm

**Pull request merged**